plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts

```diff
@@ -19,255 +19,138 @@ import {
   stringifyEntityRef,
   UserEntity,
 } from '@backstage/catalog-model';
-import * as MicrosoftGraph from '@microsoft/microsoft-graph-types';
+import { User } from '@microsoft/microsoft-graph-types';
 import limiterFactory from 'p-limit';
 import { Logger } from 'winston';
 import { MicrosoftGraphClient } from './client';
 import {
-  MICROSOFT_EMAIL_ANNOTATION,
   MICROSOFT_GRAPH_GROUP_ID_ANNOTATION,
   MICROSOFT_GRAPH_TENANT_ID_ANNOTATION,
   MICROSOFT_GRAPH_USER_ID_ANNOTATION,
 } from './constants';
-import { normalizeEntityName } from './helper';
+import {
+  defaultGroupTransformer,
+  defaultOrganizationTransformer,
+  defaultUserTransformer,
+} from './defaultTransformers';
 import { buildMemberOf, buildOrgHierarchy } from './org';
 import {
   GroupTransformer,
   OrganizationTransformer,
+  QueryMode,
   UserTransformer,
 } from './types';
 
-/**
- * The default implementation of the transformation from a graph user entry to
- * a User entity.
- *
- * @public
- */
-export async function defaultUserTransformer(
-  user: MicrosoftGraph.User,
-  userPhoto?: string,
-): Promise<UserEntity | undefined> {
-  if (!user.id || !user.displayName || !user.mail) {
-    return undefined;
-  }
-
-  const name = normalizeEntityName(user.mail);
-  const entity: UserEntity = {
-    apiVersion: 'backstage.io/v1alpha1',
-    kind: 'User',
-    metadata: {
-      name,
-      annotations: {
-        [MICROSOFT_EMAIL_ANNOTATION]: user.mail!,
-        [MICROSOFT_GRAPH_USER_ID_ANNOTATION]: user.id!,
-      },
-    },
-    spec: {
-      profile: {
-        displayName: user.displayName!,
-        email: user.mail!,
-
-        // TODO: Additional fields?
-        // jobTitle: user.jobTitle || undefined,
-        // officeLocation: user.officeLocation || undefined,
-        // mobilePhone: user.mobilePhone || undefined,
-      },
-      memberOf: [],
-    },
-  };
-
-  if (userPhoto) {
-    entity.spec.profile!.picture = userPhoto;
-  }
-
-  return entity;
-}
+const PAGE_SIZE = 999;
 
 export async function readMicrosoftGraphUsers(
   client: MicrosoftGraphClient,
   options: {
-    queryMode?: 'basic' | 'advanced';
+    queryMode?: QueryMode;
     userFilter?: string;
     userExpand?: string;
     userSelect?: string[];
     transformer?: UserTransformer;
-    logger: Logger;
+    logger?: Logger;
   },
 ): Promise<{
   users: UserEntity[]; // With all relations empty
 }> {
-  const users: UserEntity[] = [];
-  const limiter = limiterFactory(10);
+  options.logger?.info('Loading users by filter');
```

Comment:
Not very useful

Reply:
I added this from the perspective of supporting a lot of people using usign the plugin.  There are two different routes for loading users - by a filter, or by group membership.  The idea of this log (and the corresponding one on line 101 in readMicrosoftGraphUsersInGroups) was to clearly state which route we're going down.

If you're still not convinced of thier use, happy to remove, but we shoudl also remove the equilivant log in `readMicrosoftGraphUsersInGroups` for consistency